content/docs/demos/tcp_traffic_routing.md

```diff
@@ -0,0 +1,142 @@
+
+- Kubernetes cluster running Kubernetes v1.19.0 or greater.
+- Have OSM installed.
+- Have `kubectl` available to interact with the API server.
+- Have `osm` CLI available for managing the service mesh.
+
+
+## Demo
+
+The following demo shows a TCP client sending data to a `tcp-echo` server, which then echoes back the data to the client over a TCP connection.
+
+1. Set the namespace where OSM is installed.
+    ```bash
+    osm_namespace=osm-system  # Replace osm-system with the namespace where OSM is installed if different
+    ```
+
+1. Deploy the `tcp-echo` service in the `tcp-demo` namespace. The `tcp-echo` service runs on port `9000` with the `appProtocol` field set to `tcp`, which indicates to OSM that TCP routing must be used for traffic directed to the `tcp-echo` service on port `9000`.
+    ```bash
+    # Create the tcp-demo namespace
+    kubectl create namespace tcp-demo
+
+    # Add the namespace to the mesh
+    osm namespace add tcp-demo
+
+    # Deploy the service
+    kubectl apply -f https://raw.githubusercontent.com/openservicemesh/osm/main/docs/example/manifests/apps/tcp-echo.yaml -n tcp-demo
+    ```
+
+    Confirm the `tcp-echo` service and pod is up and running.
+
+    ```console
+    $ kubectl get svc,po -n tcp-demo
+    NAME               TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)    AGE
+    service/tcp-echo   ClusterIP   10.0.216.68   <none>        9000/TCP   97s
+
+    NAME                            READY   STATUS    RESTARTS   AGE
+    pod/tcp-echo-6656b7c4f8-zt92q   2/2     Running   0          97s
+    ```
+
+1. Deploy the `curl` client into the `curl` namespace.
+
+    ```bash
+    # Create the curl namespace
+    kubectl create namespace curl
+
+    # Add the namespace to the mesh
+    osm namespace add curl
+
+    # Deploy curl client in the curl namespace
+    kubectl apply -f https://raw.githubusercontent.com/openservicemesh/osm/main/docs/example/manifests/samples/curl/curl.yaml -n curl
+    ```
+
+    Confirm the `curl` client pod is up and running.
+
+    ```console
+    $ kubectl get pods -n curl
+    NAME                    READY   STATUS    RESTARTS   AGE
+    curl-54ccc6954c-9rlvp   2/2     Running   0          20s
+    ```
+
+### Using Permissive Traffic Policy Mode
+
+We will enable service discovery using [permissive traffic policy mode](/docs/guides/traffic_management/permissive_mode), which allows application connectivity to be established without the need for explicit SMI policies.
+
+1. Enable permissive traffic policy mode
+    ```bash
+    kubectl patch meshconfig osm-mesh-config -n "$osm_namespace" -p '{"spec":{"traffic":{"enablePermissiveTrafficPolicyMode":true}}}' --type=merge
+    ```
+
+1. Confirm the `curl` client is able to send and receive a response from the `tcp-echo` service using TCP routing.
+    ```console
+    $ kubectl exec -n curl -ti "$(kubectl get pod -n curl -l app=curl -o jsonpath='{.items[0].metadata.name}')" -c curl -- sh -c 'echo hello | nc tcp-echo.tcp-demo 9000'
```

Comment:
No need for interactive terminal options (`-it`) here. Won't cause an issue in this case but [generally speaking it should only be used when necessary](https://trstringer.com/kubectl-tty-issues/).